src/puppetlabs/puppetdb/catalogs.clj

```diff
@@ -114,7 +115,8 @@
    :version s/Str
    :environment (s/maybe s/Str)
    :transaction_uuid (s/maybe s/Str)
-   :producer_timestamp (s/maybe pls/Timestamp)
+   :producer_timestamp (s/pred #(not (nil? (to-timestamp %))) 'coercable-to-timestamp)
```

Comment:
I disagree that that's the point -- the point in my mind is to very specifically specify valid inputs and throw an error when the input is unexpected. By specifying pls/Timestamp we get the benefit of *knowing* that it's a timestamp. When we do stuff like this we need to look at to-timestamp to figure out what's valid.

As it happens, this schema will accept an integer, which should not be the case.